docs/contributing.md

```diff
@@ -2,9 +2,58 @@
 
 Thank you for your interest!
 
-## Running tests
-* `python -m pip install pytest`
-* `python -m pytest tests`
+## Setting up the dev environment
+
+It is ideal to create a virtual environment whenever you want to work on
+a python project. **Note:** This is not a necessity to work on sanic.
+
+Create a a new virtual environment if you dont have one, using:
+
+`python3 -m venv <any name>`
+
+Enable it using:
+
+`. <any name>/bin/activate`
+
+Get the source of sanic:
+
+`git clone https://github.com/channelcat/sanic`
+
+`cd sanic`
+
+Install sanic :
+
+`python setup.py install`
+
+Install the dev dependencies:
+
+`pip install -r requirements-dev.txt`
+
+Create a new branch for your bugfix or feature:
+
+`git checkout -b <branch name>`
+
+Now you can start working on sanic.
+
+## Tests
+
+Install pytest and flake8:
+
+`pip install pytest`
+
+`pip install flake8`
+
+Run the tests:
```

Comment:
Possible, but i felt its better if the developer knows what exactly we are using for the tests.